docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+# zedctr
+
+[zedctr](../pkg/zedctr), or "Zededa Container", is a catch-all container for many services that run as persistent services on the live rootfs. It is defined in the `services` section of the [rootfs.yml config file](../images/rootfs/yml). Ideally, most of these services would be separated out, each into its own service.
+
+This document describes the structure of `zedctr`, what services it provides and how it is built.
+
+## Location
+
+`zectr` is built in [pkg/zedctr](../pkg/zedctr) in this repository, although it includes certain outside packages when it is built. The resultant package is an OCI image named `zededa/zedctr`, which may be pushed to the image registry.
+
+## Runtime Config
+
+When run in the live rootfs, it runs with maximal capabilities, essentially as if it is running outside a container. These runtime capabilities are defined in [build.yml](../pkg/zedctr/build.yml).
+
+* It has `capabilities: all`, which should be split into individual capabilities listed for each service.
+* It has `pid: host`, working in the same PID namespace as the host.
+* It has `net: host`, working directly in the host networking namespace.
+* It mounts most of the root filesystem into its own namespace, including `/:/hostfs`. The mounts are listed below:
+
+```yml
+  binds:
+    - /lib/modules:/lib/modules
+    - /dev:/dev
+    - /etc/resolv.conf:/etc/resolv.conf
+    - /run:/run
+    - /var/config:/config
+    - /var/log:/var/log/dom0
+    - /:/hostfs
+```
+
+## Components
+
+The image `zededa/zedctr` is built from its [Dockerfile](../pkg/zedctr/Dockerfile), which is generated from its [Dockerfile.in](../pkg/zedctr/Dockerfile.in), with the relevant image names and tags patched in.
+
+It is built from the following packages, installed into the following locations:
+
+|Package|Image|Source|Install|Purpose|Comments|
+|---|---|---|---|---|
+|ztools|`zededa/ztools`|[go-provision](https://github.com/zededa/go-provision)|`/`|Interface with zedcloud||
+|lisp|`zededa/lisp`|[lisp](https://github.com/zededa/lisp)|[Locator/Identifier Separation Protocol](https://en.wikipedia.org/wiki/Locator/Identifier_Separation_Protocol)|see list [here](#lisp-install)|implementation for mesh networking|custom implementation in [go](https://golang.org), as the reference implementation is in python and has unneeded services|
+|xen tools|`zededa/xen-tools`|[pkg/xen-tools](../pkg/xen-tools)|`/`|Xen utilities||
+|dnsmasg|`zededa/dnsmasq`|[pkg/dnsmasq](../pkg/dnsmasq)|`/usr/sbin/dnsmasq` to `/opt/zededa/bin/dnsmasq`|local DNS server|Unclear why we need a local build|
```

Comment:
We have a patch for IPv6 in dnsmasq which has not yet been upstreamed.

Reply:
Got it. Updating. I see it now.

Reply:
Actually, it has been fixed, see http://thekelleys.org.uk/gitweb/?p=dnsmasq.git;a=blob;f=src/option.c;h=5debcbcf5e9ef65c29cf85438c152793f9884be1;hb=HEAD#l3070 

I confirmed that it is in the 2.80 release, and that 2.80 is available in `apk` in `alpine:3.9`. Marking as such.